src/main/kotlin/net/atos/zac/smartdocuments/templates/SmartDocumentsTemplateConverter.kt

```diff
@@ -127,4 +127,24 @@ object SmartDocumentsTemplateConverter {
         id = smartDocumentsTemplate.smartDocumentsId,
         name = smartDocumentsTemplate.name
     )
+
+    fun Set<RESTSmartDocumentsTemplateGroup>.toStringRepresentation(): Set<String> {
+        val result = mutableSetOf<String>()
+        this.map { result.addAll(convertTemplateGroupToStringRepresentation(it, null)) }
+        return result
+    }
+
+    private fun convertTemplateGroupToStringRepresentation(
+        group: RESTSmartDocumentsTemplateGroup,
+        parent: String?
+    ): Set<String> {
+        val thisGroupString = "group.${group.id}.${group.name}"
+        val groupString = if (parent.isNullOrEmpty()) thisGroupString else "$parent.$thisGroupString"
+        val groupRepresentation = mutableSetOf(groupString)
+
+        group.templates?.mapTo(groupRepresentation) { "$groupString.template.${it.id}.${it.name}" }
```

Comment:
A little bit hard to read this code. I think we can probably make this a bit more functional / Kotlin idiomatic. @bas-info-nl is usually very good at that kind of stuff. ;-)